pyabc/distance/scales.py

```diff
@@ -19,113 +19,134 @@
 
 Here, "only distance to observation" means that the in-sample variation
 is not taken into account.
+
+For AdaptiveAggregatedDistance, for which the data are not summary statistic
+values, but distance values, instead use either of
+* span
+* mean
+* median
 """
 
 
 import numpy as np
 
 
-def median_absolute_deviation(**kwargs):
+def median_absolute_deviation(data, **kwargs):
     """
     Calculate the sample `median absolute deviation (MAD)
     <https://en.wikipedia.org/wiki/Median_absolute_deviation/>`_
     from the median, defined as
     median(abs(data - median(data)).
     """
-    data = np.asarray(kwargs['data'])
+    data = np.array(data)
     mad = np.median(np.abs(data - np.median(data)))
     return mad
 
 
-def mean_absolute_deviation(**kwargs):
+def mean_absolute_deviation(data, **kwargs):
     """
     Calculate the mean absolute deviation from the mean.
     """
-    data = np.asarray(kwargs['data'])
+    data = np.array(data)
     mad = np.mean(np.abs(data - np.mean(data)))
     return mad
 
 
-def standard_deviation(**kwargs):
+def standard_deviation(data, **kwargs):
     """
     Calculate the sample `standard deviation (SD)
     <https://en.wikipedia.org/wiki/Standard_deviation/>`_.
     """
-    data = np.asarray(kwargs['data'])
     std = np.std(data)
     return std
 
 
-def bias(**kwargs):
+def bias(data, x_0, **kwargs):
     """
     Bias of sample to observed value.
     """
-    data = np.asarray(kwargs['data'])
-    x_0 = kwargs['x_0']
     bias = np.abs(np.mean(data) - x_0)
```

Comment:
here you could use the `mean` function defined down below (and following)

Reply:
yeah, one could do that, but since those functions are just shallow wrappers around the numpy functionality, I don't really see the reason (and minimally increases computation time). the problem those wrapper functions solve is that some of the scale functions e.g. requre x_0, some only data, so I defined a common interface by which all functions can be called, with additional input values just being ignored.

Reply:
also, calling the own function adds another layer of uncertainty, because one has to make sure that the code in those functions does not change in some way. numpy in contrast will definitely not change.